core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -228,7 +236,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     val partitionsRemaining = controller.shutdownBroker(controlledShutdownRequest.brokerId)
     val controlledShutdownResponse = new ControlledShutdownResponse(controlledShutdownRequest.correlationId,
       Errors.NONE, partitionsRemaining)
-    requestChannel.sendResponse(new Response(request, new RequestOrResponseSend(request.connectionId, controlledShutdownResponse)))
+    sendResponseExemptThrottle(request, new Response(request, new RequestOrResponseSend(request.connectionId, controlledShutdownResponse)))
```

Comment:
Yes, `authorizeClusterAction` takes care of throttling for unauthorized request and all other paths including error path goes through this `sendResponseExemptThrottle` which does not perform throttling.